src/Transfer.js

```diff
@@ -0,0 +1,390 @@
+import './Transfer/types.js'
+import React, { Children, useState } from 'react'
+import propTypes from '@dhis2/prop-types'
+
+import { Actions } from './Transfer/Actions.js'
+import { AddAll } from './Transfer/AddAll.js'
+import { AddIndividual } from './Transfer/AddIndividual.js'
+import { Container } from './Transfer/Container.js'
+import { Filter } from './Transfer/Filter.js'
+import { LeftFooter } from './Transfer/LeftFooter.js'
+import { LeftHeader } from './Transfer/LeftHeader.js'
+import { LeftSide } from './Transfer/LeftSide.js'
+import { PickedOptions } from './Transfer/PickedOptions.js'
+import { RemoveAll } from './Transfer/RemoveAll.js'
+import { RemoveIndividual } from './Transfer/RemoveIndividual.js'
+import { ReorderingActions } from './Transfer/ReorderingActions.js'
+import { RightFooter } from './Transfer/RightFooter.js'
+import { RightSide } from './Transfer/RightSide.js'
+import { SourceOptions } from './Transfer/SourceOptions.js'
+import {
+    addAllSelectableSourceOptions,
+    addIndividualSourceOptions,
+    addKeyToReactOptions,
+    createDoubleClickHandlers,
+    extractPickedReactOptions,
+    filterOptions as defaultFilterOptions,
+    filterOutOptions,
+    getSubsetByFilter,
+    isReorderDownDisabled,
+    isReorderUpDisabled,
+    moveHighlightedPickedOptionDown,
+    moveHighlightedPickedOptionUp,
+    removeAllPickedOptions,
+    removeIndividualPickedOptions,
+    useHighlightedOptions,
+} from './Transfer/helper/index.js'
+import {
+    singleSelectedPropType,
+    multiSelectedPropType,
+} from './common-prop-types.js'
+
+/**
+ * @module
+ * @param {Transfer.PropTypes} props
+ *
+ * @returns {React.Component}
+ *
+ * @example import { Transfer } from @dhis2/ui-core
+ * @see Specification: {@link https://github.com/dhis2/design-system/blob/master/organisms/transfer.md|Design system}
+ * @see Live demo: {@link /demo/?path=/story/transfer--basic|Storybook}
+ */
+export const Transfer = ({
+    onChange,
+
+    children,
+    className,
+    dataTest,
+    disabled,
+    emptyOptionsPlaceholder,
+    emptySelectionPlaceholder,
+    enableOrderChange,
+    filter,
+    filterLabel,
+    filterOptions,
+    height,
+    initialFilter,
+    labelAddAll,
+    labelAddIndividual,
+    labelRemoveAll,
+    labelRemoveIndivitual,
+    leftFooter,
+    leftHeader,
+    maxSelections,
+    optionsWidth,
+    rightFooter,
+    selected,
+    selectedWidth,
+    showFilterInput,
+    onFilterChange,
+}) => {
+    /**
+     * As this component will filter and sort children
+     * in the selected area, adding keys will prevent
+     * reconciliation from suffering badly
+     */
+    const allOptions = addKeyToReactOptions(children)
```

Comment:
I see that in the Transfer story the options already have keys (see [here](https://github.com/dhis2/ui-core/blob/71432d60877f0de4ebca20eacd2e49d2e1f691dc/src/__demo__/Transfer.stories.js#L31)). Given that React will also [warn for list items without keys](https://reactjs.org/docs/lists-and-keys.html#basic-list-component) and assign keys itself if they're missing (as a last resort), I think it would be more idiomatic to defer this responsibility to the dev implementing this component in an app. Unless I'm misreading sth here.

Reply:
The reason I did this is that it will work with static children as well.
But a `Transfer` with a lot of static options is probably a never occurring edge-case, we could get rid of this. Do you prefer this to be removed?

Reply:
Yeah, I'd personally remove it. It seems more idiomatic.

Reply:
Also vote to remove it.